mhkit/wave/io/hindcast.py

```diff
@@ -216,13 +218,69 @@ def request_wpto_directional_spectrum(lat_lon, year, tree=None,
     waveKwargs = {'tree':tree,'unscale':unscale,'str_decode':str_decode, 'hsds':hsds}
         
     with WaveX(wave_path, **waveKwargs) as rex_waves:
-        # get data
-        data_raw = rex_waves.get_lat_lon_df(parameter,lat_lon)
-        # get metadata
-        col = data_raw.columns[:]
-        meta = rex_waves.meta.loc[col,:]
-        meta = meta.reset_index(drop=True) 
+        # Get graphical identifier
+        gid = rex_waves.lat_lon_gid(lat_lon)  
+        
+        # Setup index and columns
+        if isinstance(gid, (int, np.integer)):
+            columns = [gid]
+        else:
+            columns = gid   
+        time_index = rex_waves.time_index
+        frequency = rex_waves['frequency']
+        direction = rex_waves['direction']
+        index = pd.MultiIndex.from_product(
+            [time_index, frequency, direction],
+            names=['time_index', 'frequency', 'direction']
+        )                    
+
+        # Create bins for multiple smaller API dataset requests
+        N=6
+        length = len(rex_waves)        
+        quotient=length//N
+        remainder=length%N    
+        bins = [i*quotient for i in range(N+1) ]
+        bins[-1]+= remainder        
+        index_bins = (np.array(bins)*len(frequency)*len(direction)).tolist()
+        
+        # Request multiple datasets and add to dictionary
+        datas={}
+        for i in range(len(bins)-1):
+            idx=index[index_bins[i]:index_bins[i+1]] 
+
+            # Request with exponential back off wait time
+            sleep_time = 2
+            num_retries = 4
+            for x in range(0, num_retries):  
```

Comment:
@akeeste the sleep time should only increase by the number of retries which is currently 4. Which would lead to a total max time of 2^4 seconds of wait time. Does this address your comment?

Reply:
@ssolson yes that does, thanks. Missed that on my first read